### PR TITLE
Add configurable ROI margin

### PR DESCRIPTION
**Pull request by MatthijsBurgh**

### CODE REVIEW

File: ed_perception/src/perception_plugin_image_recognition.cpp [hunk 1]
```diff
@@ -105,16 +108,20 @@ bool PerceptionPluginImageRecognition::srvClassify(ed_perception_msgs::Classify:
...
             p_min.y = std::min(p_min.y, p.y);
             p_max.x = std::max(p_max.x, p.x);
             p_max.y = std::max(p_max.y, p.y);
         }
-        cv::Rect roi = cv::Rect(std::min(p_min.x + 5, image.cols),
-                                std::min(p_min.y + 5, image.rows),
-                                std::max(p_max.x - p_min.x - 5, 0),
-                                std::max(p_max.y - p_min.y - 5, 0));
+        
+        int x = std::max(std::min(p_min.x - roi_margin_, image.cols), 0);
```
Comment (PetervDooren) [inline ed_perception/src/perception_plugin_image_recognition.cpp:118]: These statements only need a max or a min. We know p_min and p_max lie within the image and margin >0

File: ed_perception/src/perception_plugin_image_recognition.cpp [hunk 2]
```diff
@@ -31,7 +31,7 @@ namespace perception
 
 // ----------------------------------------------------------------------------------------------------
 
-PerceptionPluginImageRecognition::PerceptionPluginImageRecognition()
+PerceptionPluginImageRecognition::PerceptionPluginImageRecognition() : roi_margin_(0)
```
Comment (PetervDooren) [inline ed_perception/src/perception_plugin_image_recognition.cpp:34]: Does this initialise roi margin to 0?
Preferably set it to 5, the default value

**Review by PetervDooren — CHANGES_REQUESTED** [review]

File: ed_perception/src/perception_plugin_image_recognition.cpp (hunk 2 above)
Comment (MatthijsBurgh) [reply ed_perception/src/perception_plugin_image_recognition.cpp:34]: In the initialize it is set to 5, before checking if it in tbe config.

File: ed_perception/src/perception_plugin_image_recognition.cpp (hunk 1 above)
Comment (MatthijsBurgh) [reply ed_perception/src/perception_plugin_image_recognition.cpp:118]: No, margin can also be negative. Therefore `p_min-margin` can be outside the image. So both `min`/`max` are needed.

**Comment by MatthijsBurgh** [comment]
@PetervDooren please check my responses to your questions

**Review by PetervDooren — APPROVED** [review]

**Pull request merged**